blocks/monitoring/aggregation.py

```diff
@@ -138,11 +138,62 @@ def get_aggregator(self):
         return aggregator
 
 
-def mean(numerator, denominator=1.):
+def mean(numerator, denominator=1., name=None):
     """Mean of quantity (numerator) over a number (denominator) values."""
     variable = numerator / denominator
     variable.tag.aggregation_scheme = Mean(numerator, denominator)
-    variable.name = numerator.name
+    if name is not None:
```

Comment:
Without opening a can of worms by asking ourselves whether or not `variable.copy()` should preserve `variable.tag`, we could just use a new utility function `rename` instead of `named_copy` that would change the name of its argument and return it. Moreover, it is a shame that `setattr` doesn't  return its first operand, otherwise we could just write `setattr(mean(x, y), 'name', 'per-token-cost')`.